src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +192,55 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.push(
+      `?hazardLayerId=${hazardLayerId}&baselineLayerId=${baselineLayerId}&date=${selectedDate}&statistic=${statistic}&aboveThreshold=${aboveThreshold}&belowThreshold=${belowThreshold}`,
+    );
   };
 
+  useEffect(() => {
+    const queryString = history.location.search;
+    const params = new URLSearchParams(queryString);
+    const hazardLayerParamId = params.get('hazardLayerId');
+    const baselineLayerParamId = params.get('baselineLayerId');
+    const selectedParamDate: number = Number(params.get('date'));
+    const statisticParam = params.get('statistic');
+    const aboveThresholdParam: string = params.get('aboveThreshold') || '';
+    const belowThresholdParam: string = params.get('belowThreshold') || '';
+
+    if (!hazardLayerParamId || !baselineLayerParamId || !statisticParam) {
+      return;
+    }
+    function isHazardLayerId(layer: string): layer is LayerKey {
+      return typeof layer === 'string';
+    }
+    function isBaselineLAyerId(layer: string): layer is LayerKey {
+      return typeof layer === 'string';
+    }
+
+    if (isHazardLayerId(hazardLayerParamId)) {
+      setHazardLayerId(hazardLayerParamId);
+    }
+    if (isBaselineLAyerId(baselineLayerParamId)) {
+      setBaselineLayerId(baselineLayerParamId);
+    }
+
+    // Mean is selected by default
+    if (statisticParam === 'median') {
+      setStatistic(AggregationOperations.Median);
+    }
+
+    setSelectedDate(selectedParamDate);
+    setAboveThreshold(aboveThresholdParam);
+    setBelowThreshold(belowThresholdParam);
+
+    // Avoid Running Analyser if boundary Layer data is Null
+    if (boundaryLayerData && hazardLayerId && baselineLayerId && selectedDate) {
+      runAnalyser();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [boundaryLayerData]);
```

Comment:
We shouldn't aim for this code to just be run on startup. If the url changes because it is manually edited, or the user goes back a page, the page should change accordingly.

Reply:
This takes care of all the scenarios you mentioned.
If the user manually edited the URL, the page is going to rerender everything and that will execute `runAnalyser`. The same thing happens when users navigate backs.

Reply:
I tried navigating back just now, but nothing happened. Which browser did you test on? (I used Edge)

We shouldn't ignore this eslint rule anyway, as its a paradigm of `useEffect` to exhaust all dependencies in use. Adding all dependencies will also add back support for all browsers.